torch/_inductor/ir.py

```diff
@@ -3630,21 +3630,42 @@ def __init__(
 
         self.op_overload = kernel
 
-        op_overload_packet = (
-            kernel._overloadpacket
-            if isinstance(kernel, torch._ops.OpOverload)
-            else kernel
-        )
+        # TODO: Need to revisit schema matching to find the correct OpOverload from OpOverloadPacket
+        assert isinstance(
+            kernel,
+            (
+                torch._ops.OpOverload,
+                torch._ops.OpOverloadPacket,
+                torch._ops.HigherOrderOperator,
+            ),
+        ), f"Fails to create FallbackKernel for {kernel}: {type(kernel)} not supported"
 
-        if (
-            getattr(torch.ops.aten, op_overload_packet.__name__, None)
-            is op_overload_packet
-        ):
-            self.kernel = (
-                f"at::{op_overload_packet.__name__}"
-                if V.graph.cpp_wrapper
-                else f"aten.{kernel.__name__}"
+        if kernel.__module__ == "torch._ops.aten":
+            op_base_name = (
+                kernel.__name__.split(".")[0]
+                if isinstance(kernel, torch._ops.OpOverload)
+                else kernel.__name__
             )
+            if V.graph.cpp_wrapper:
+                if (
+                    isinstance(kernel, torch._ops.OpOverload)
+                    and kernel._overloadname != "default"
+                ):
+                    # Calling with the default kernel name can lead to ambiguous behavior like the following example.
+                    # repeat_interleave(const at::Tensor & repeats, c10::optional<int64_t> output_size=c10::nullopt)
+                    # repeat_interleave(const at::Tensor & self, int64_t repeats,
+                    #       c10::optional<int64_t> dim=c10::nullopt, c10::optional<int64_t> output_size=c10::nullopt)
+                    #
+                    # In theory, we should be able to do the following codegen for all aten fallback ops, but more
+                    # plumbing needs to be done in order to make that happen.
+                    # TODO: follow up on this
```

Comment:
Do you mean you want to do this for OpOverloadPacket in addition to OpOverload, or also OpOverload that are the "default" overload?

If it is the latter, you can just move the kernel._overloadname thing into the true branch: 
```
kernel_name = kernel.__name__.replace('.', '_') if kernel._overloadname != "default" else kernel.__name__.replace('.default', '')
self.kernel = f"at::_ops::{kernel_name}::call"
```

Reply:
No, I meant putting `schema = kernel._schema` into the else branch will cause some other tests to fail, which I will fix as a follow up.